### PR TITLE
Fetch source maps over HTTP

### PR DESCRIPTION
**Pull request by shishkin**
I was running into problems with source map URLs pointing to remote https:// resources. Same problem as #77.

This fix seems to solve my problem by fetching the remote source maps for future bundling. Ideally I would prefer source-map-loader to ignore HTTP URLs and pass them verbatim for the browser to fetch remote source maps when needed. However I couldn't figure out how to do that.

Any feedback is welcome!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -68,6 +69,20 @@ module.exports = function(input, inputMap) {
 	function untouched() {
 		callback(null, input, inputMap);
 	}
+	function fetchHttp(source, callback) {
+		http.get(source, function (resp) {
```
Comment (felixfbecker) [inline index.js]: Built-in `http.get()` is very bare-bones, e.g. doesn't follow redirects. Should probably use some higher-level library.
Also, does this need caching? Or is there already a cache on another layer?

**Comment by Atry** [comment]
This PR does not handle HTTP redirection.

If the original source map to https://github.com/ThoughtWorksInc/Binding.scala/raw/78443c86e5046bd1e13f1504fa9280e668d55eb9/Binding/src/main/scala/com/thoughtworks/binding/Binding.scala, this PR will create a file with the following content:

```
<html><body>You are being <a href="https://raw.githubusercontent.com/ThoughtWorksInc/Binding.scala/78443c86e5046bd1e13f1504fa9280e668d55eb9/Binding/src/main/scala/com/thoughtworks/binding/Binding.scala">redirected</a>.</body></html>


// WEBPACK FOOTER //
// https://github.com/ThoughtWorksInc/Binding.scala/raw/78443c86e5046bd1e13f1504fa9280e668d55eb9/Binding/src/main/scala/com/thoughtworks/binding/Binding.scala
```

File: index.js (hunk 1 above)
Comment (evilebottnawi) [reply index.js]: you can use `got`

Comment (shishkin) [reply index.js]: > Built-in `http.get()` is very bare-bones, e.g. doesn't follow redirects. Should probably use some higher-level library.
> Also, does this need caching? Or is there already a cache on another layer?

I understand. Any considerations about bringing in a new dependency?

Comment (shishkin) [reply index.js]: > you can use `got`

I'll try. Thanks for recommendation.

**Comment by evilebottnawi** [comment]
Close in favor https://github.com/webpack-contrib/source-map-loader/pull/113, will be implemented for `1.1.0` version, anyway sorry for delay and thanks for the PR

**Pull request closed without merging**